pkg/i2gw/providers/gce/gce_extensions_test.go

```diff
@@ -0,0 +1,166 @@
+				Annotations: map[string]string{
+					annotations.BetaBackendConfigKey: `{"default":"betabackendconfig-1"}`,
+				},
+			},
+		},
+	}
+	backendConfigs := map[types.NamespacedName]*backendconfigv1.BackendConfig{
+		{Namespace: testNamespace, Name: testBeConfigName}: {},
+	}
+	betaBackendConfigs := map[types.NamespacedName]*backendconfigv1beta1.BackendConfig{
+		{Namespace: testNamespace, Name: testBetaBeConfigName}: {},
+	}
+
+	provider := NewProvider(&i2gw.ProviderConf{})
+	gceProvider := provider.(*Provider)
+	gceProvider.storage = newResourcesStorage()
+	gceProvider.storage.Services = services
+	gceProvider.storage.BackendConfigs = backendConfigs
+	gceProvider.storage.BetaBackendConfigs = betaBackendConfigs
+
+	beConfigToSvcs := getBackendConfigMapping(context.TODO(), gceProvider.storage)
+
+	beConfigSvcList := beConfigToSvcs[types.NamespacedName{Namespace: testNamespace, Name: testBeConfigName}]
+	if len(beConfigSvcList) != 1 ||
+		beConfigSvcList[0].Namespace != testNamespace ||
+		beConfigSvcList[0].Name != testServiceName1 {
+		t.Errorf("Got BackendConfig %s mapped to %s, expected [%s/%s]", testBeConfigName, beConfigSvcList, testNamespace, testServiceName1)
+	}
+
+	betaBeConfigSvcList := beConfigToSvcs[types.NamespacedName{Namespace: testNamespace, Name: testBetaBeConfigName}]
+	if len(betaBeConfigSvcList) != 1 ||
+		betaBeConfigSvcList[0].Namespace != testNamespace ||
+		betaBeConfigSvcList[0].Name != testServiceName2 {
+		t.Errorf("Got BackendConfig %s mapped to %s, expected [%s/%s]", testBetaBeConfigName, betaBeConfigSvcList, testNamespace, testServiceName2)
+	}
+}
+
+func TestGetBackendConfigName(t *testing.T) {
+	t.Parallel()
+
+	testNamespace := "test-namespace"
+	testServiceName := "test-service"
+	testBeConfigName := "backendconfig-1"
+
+	testCases := []struct {
+		desc           string
+		service        *apiv1.Service
+		beConfigKey    string
+		expectedName   string
+		expectedExists bool
+	}{
+		{
+			desc: "Service with v1 BackendConfig, using default Config over all ports",
+			service: &apiv1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testServiceName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						annotations.BackendConfigKey: `{"default":"backendconfig-1"}`,
+					},
+				},
+			},
+			beConfigKey:    annotations.BackendConfigKey,
+			expectedName:   testBeConfigName,
+			expectedExists: true,
+		},
+		{
+			desc: "Service with v1beta1 BackendConfig, using default Config over all ports",
+			service: &apiv1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testServiceName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						annotations.BetaBackendConfigKey: `{"default":"backendconfig-1"}`,
+					},
+				},
+			},
+			beConfigKey:    annotations.BetaBackendConfigKey,
+			expectedName:   testBeConfigName,
+			expectedExists: true,
+		},
+		{
+			desc: "Service with v1 BackendConfig, using Port Config, not supported",
```

Comment:
Not really any great solution here until we can support ports in targetRefs, but it may be slightly less bad to just pick the first one and translate that, combined with a warning describing what happened.

Reply:
Sure just updated to use the one associated with alphabetically lowest port so we get a consistent behavior.